src/functions.php

```diff
@@ -432,6 +432,18 @@ function is_valid_hostname($hostname)
     );
 }
 
+/**
+ * Checks if supplied parameter is a valid host label
+ *
+ * @param $label
+ * @return bool
+ */
+function is_valid_hostlabel($label)
+{
+    return preg_match("/^([a-z\d](-*[a-z\d])*)$/", $label)
+        && preg_match("/^.{1,63}$/", $label);
```

Comment:
Go is using a different regex: "[a-zA-Z0-9-]{1,63}"
Is there anywhere we could find agreed upon rules for the regex?

Reply:
ruby is using this: /^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{,63}(?<!-)$/

Reply:
Follow up: after asking everyone, it seems like there's no standard agreed upon option, so this works great :) hopefully in the future there will be one regex that every team implements

Reply:
Chiming in here @SamRemis since I know you brought it up earlier, https://github.com/aws/aws-sdk-go/blob/master/aws/endpoints/v3model.go#L10 is the specific regex that we are using to validate this functionality.

Reply:
Updated to use a newly proposed regex. Will wait until cross-sdk test cases are established and update with those as well.